santorini-common/src/objects/state.rs

```diff
@@ -50,10 +51,23 @@ impl State {
     // }
 
     /// Mutates the game state. Any errors surfaced here are not irrecoverable.
-    fn apply_command(&mut self, command: &dyn Command) -> Result<(), SantoriniError> {
+    pub fn apply_command(&mut self, command: &dyn Command) -> Result<(), SantoriniError> {
         command.execute(self)?;
         Ok(())
     }
+
+    // Returns a winner, assuming that a legal state has exactly one winner
+    pub fn winner(&self) -> Option<Player> {
+        for space in self.board.iter().flatten() {
+            if let Some(worker) = space.worker() {
+                if space.tower().level() == Level::Three {
+                    return Some(worker.player());
```

Comment:
The way you'd structure this code is by using `Iterator::find`. This returns the first instance of an element that satisfies a given predicate, or otherwise None.

Reply:
I played around with this. `find` doesn't really help the code here. If I were to e.g. find for a "winning" space, I would then need a second bit to destructure the space again to get the player. I used `find_map` instead to do it directly, but then the expression is pretty similar. 

I also considered `filter` but then you do excess work.

Reply:
`find_map` sgtm. You can modify the inner part to use the combinator `space.worker().and_then(Worker::player)` . Monads in action wow!

Reply:
A side note is that we try to avoid using `return` as much as possible